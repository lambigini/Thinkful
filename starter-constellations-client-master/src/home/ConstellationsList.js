import React from "react";
import Constellation from "./Constellation";
import NoConstellationsMessage from "./NoConstellationsMessage";

function ConstellationsList ({ constellations }) {
  if (!constellations.length) return <NoConstellationsMessage />;

  const cards = constellations.map((constellation) => (
    <Constellation key={constellation.id} constellation={constellation} />
  ));
  return (
    <div className="container">
      <div className="row">{cards}</div>
    </div>
  );
};

export default ConstellationsList;
