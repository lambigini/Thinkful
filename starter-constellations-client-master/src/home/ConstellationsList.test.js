import React from "react";
import ConstellationsList from "./ConstellationsList";
import renderer from "react-test-renderer";

it("renders a single constellation card when provided data", () => {
  const constellations = [
    {
      id: "zb8QvVt",
      name: "Crater",
      meaning: "Cup",
      starsWithPlanets: 10,
      quadrant: "SQ2",
    },
  ];
  const component = renderer.create(
    <ConstellationsList constellations={constellations} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it("renders multiple constellation cards when provided data", () => {
  const constellations = [
    {
      id: "zvBpkNq",
      name: "Gemini",
      meaning: "Twins",
      starsWithPlanets: 11,
      quadrant: "NQ2",
    },
    {
      id: "UPtAzfV",
      name: "Hydrus",
      meaning: "Water Snake",
      starsWithPlanets: 5,
      quadrant: "SQ1",
    },
    {
      id: "ISlkF4G",
      name: "Leo",
      meaning: "Lion",
      starsWithPlanets: 19,
      quadrant: "NQ2",
    },
  ];
  const component = renderer.create(
    <ConstellationsList constellations={constellations} />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
