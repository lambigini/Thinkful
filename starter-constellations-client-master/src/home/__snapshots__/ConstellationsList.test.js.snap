// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single constellation card when provided data 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="card col-12 col-md-4 rounded-0"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Crater
           -
           
          <small
            className="text-muted"
          >
            Cup
          </small>
        </h5>
        <p
          className="card-text"
        >
          <strong>
            Stars with Planets: 
          </strong>
          10
        </p>
        <p
          className="card-text"
        >
          <strong>
            Quadrant: 
          </strong>
          SQ2
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders multiple constellation cards when provided data 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="card col-12 col-md-4 rounded-0"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Gemini
           -
           
          <small
            className="text-muted"
          >
            Twins
          </small>
        </h5>
        <p
          className="card-text"
        >
          <strong>
            Stars with Planets: 
          </strong>
          11
        </p>
        <p
          className="card-text"
        >
          <strong>
            Quadrant: 
          </strong>
          NQ2
        </p>
      </div>
    </div>
    <div
      className="card col-12 col-md-4 rounded-0"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Hydrus
           -
           
          <small
            className="text-muted"
          >
            Water Snake
          </small>
        </h5>
        <p
          className="card-text"
        >
          <strong>
            Stars with Planets: 
          </strong>
          5
        </p>
        <p
          className="card-text"
        >
          <strong>
            Quadrant: 
          </strong>
          SQ1
        </p>
      </div>
    </div>
    <div
      className="card col-12 col-md-4 rounded-0"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Leo
           -
           
          <small
            className="text-muted"
          >
            Lion
          </small>
        </h5>
        <p
          className="card-text"
        >
          <strong>
            Stars with Planets: 
          </strong>
          19
        </p>
        <p
          className="card-text"
        >
          <strong>
            Quadrant: 
          </strong>
          NQ2
        </p>
      </div>
    </div>
  </div>
</div>
`;
